File: RailgunVB/Core/Api/Youtube/YoutubeVideoData.vb
Namespace Core.Api.Youtube
    
    Public Class YoutubeVideoData
    
        Public ReadOnly Property Name As String
        Public ReadOnly Property Id As String
        Public ReadOnly Property Uploader As String
        
        Friend Sub New(name As String, id As String, uploader As String)
            Me.Name = name
            Me.Id = id
            Me.Uploader = uploader
        End Sub
        
    End Class
    
End NameSpace

File: Railgun/Apis/Youtube/YoutubeVideoData.cs
namespace Railgun.Apis.Youtube
{
    public class YoutubeVideoData
    {
        public string Id { get; }
        public string Name { get; }
        public string Uploader { get; }
        public string Url
        { 
            get
            {
                return string.Format("https://youtu.be/{0}", Id);
            }
        }
        
        internal YoutubeVideoData(string name, string id, string uploader)
        {
            Id = id;
            Name = name;
            Uploader = uploader;
        }
    }
}